elasticsearch/client.go

```diff
@@ -126,12 +133,68 @@ func newV7Client(apikey, user, pwd string, addresses []string, transport http.Ro
 	})
 }
 
-func newV8Client(apikey, user, pwd string, addresses []string, transport http.RoundTripper) (*v8.Client, error) {
-	return v8.NewClient(v8.Config{
+func newV8Client(apikey, user, pwd string, addresses []string, transport http.RoundTripper) (*esv8.Client, error) {
+	return esv8.NewClient(esv8.Config{
 		APIKey:    apikey,
 		Username:  user,
 		Password:  pwd,
 		Addresses: addresses,
 		Transport: transport,
 	})
 }
+
+type JSONResponse struct {
+	content io.ReadCloser
+	err     error
+}
+
+func (r JSONResponse) DecodeTo(i interface{}) error {
+	if r.err != nil {
+		return r.err
+	}
+	defer r.content.Close()
+	err := json.NewDecoder(r.content).Decode(&i)
+	return err
+}
+
+// each header has the format "key: value"
+func makeRequest(method, path string, body interface{}, headers ...string) (*http.Request, error) {
```

Comment:
Because of convenience... In go is possible to express 0..N arguments, but not 0..1
if I require `Header`, many callers will have to pass an argument they don't care about (so, `nil`); and if I require `...Header`, I will need a nested loop to iterate over each key of each map, even when in practice there will only be 0 or 1 Header maps.
This seemed expressive and straightforward.